frontend_app_thread/src/container/PopupCreateThread.jsx

```diff
@@ -6,54 +6,74 @@ import {
   handleFetchResult,
   CUSTOM_EVENT,
-  buildHeadTitle
+  buildHeadTitle,
+  appContentFactory,
+  TLM_CORE_EVENT_TYPE as TLM_CET,
+  TLM_ENTITY_TYPE as TLM_ET,
+  TLM_SUB_TYPE as TLM_ST,
+  TracimComponent
 } from 'tracim_frontend_lib'
 import { postThreadContent } from '../action.async.js'
-// FIXME - GB - 2019-07-04 - The debug process for creation popups are outdated
-// https://github.com/tracim/tracim/issues/2066
 import { debug } from '../debug.js'
 
 class PopupCreateThread extends React.Component {
   constructor (props) {
     super(props)
+
+    const param = props.data || debug
+    props.setApiUrl(param.config.apiUrl)
+
     this.state = {
       appName: 'thread', // must remain 'thread' because it is the name of the react built app (which contains Threac and PopupCreateThread)
-      config: props.data ? props.data.config : debug.config,
-      loggedUser: props.data ? props.data.loggedUser : debug.loggedUser,
-      workspaceId: props.data ? props.data.workspaceId : debug.workspaceId,
-      folderId: props.data ? props.data.folderId : debug.folderId,
+      config: param.config,
+      loggedUser: param.loggedUser,
+      workspaceId: param.workspaceId,
+      folderId: param.folderId,
       newContentName: ''
     }
 
     // i18n has been init, add resources from frontend
     addAllResourceI18n(i18n, this.state.config.translation, this.state.loggedUser.lang)
     i18n.changeLanguage(this.state.loggedUser.lang)
 
-    document.addEventListener(CUSTOM_EVENT.APP_CUSTOM_EVENT_LISTENER, this.customEventReducer)
+    props.registerCustomEventHandlerList([
+      { name: CUSTOM_EVENT.ALL_APP_CHANGE_LANGUAGE, handler: this.handleAllAppChangeLanguage }
+    ])
+
+    props.registerLiveMessageHandlerList([
+      { entityType: TLM_ET.CONTENT, coreEntityType: TLM_CET.CREATED, optionalSubType: TLM_ST.THREAD, handler: this.handleContentCreated }
+    ])
   }
 
   componentDidMount () {
     this.setHeadTitle()
   }
 
-  componentWillUnmount () {
-    document.removeEventListener(CUSTOM_EVENT.APP_CUSTOM_EVENT_LISTENER, this.customEventReducer)
+  handleAllAppChangeLanguage = data => {
+    console.log('%c<PopupCreateThread> Custom event', 'color: #28a745', CUSTOM_EVENT.APP_CUSTOM_EVENT_LISTENER, data)
+
+    this.props.appContentCustomEventHandlerAllAppChangeLanguage(data, this.setState.bind(this), i18n, false)
+    this.setHeadTitle()
   }
 
-  customEventReducer = ({ detail: { type, data } }) => {
-    switch (type) {
-      case CUSTOM_EVENT.ALL_APP_CHANGE_LANGUAGE:
-        console.log('%c<PopupCreateThread> Custom event', 'color: #28a745', type, data)
-        this.setState(prev => ({
-          loggedUser: {
-            ...prev.loggedUser,
-            lang: data
-          }
-        }))
-        i18n.changeLanguage(data)
-        this.setHeadTitle()
-        break
-    }
+  handleContentCreated = data => {
+    const { state } = this
+
+    if (Number(data.content.parent_id) !== Number(state.folderId) ||
```

Comment:
is there any chance that `data.content.parent_id` or `state.folderId` is not a number?

Reply:
I asked myself the same question (it is old code) and yes, parent_id might come as a string

Reply:
/: ok!